config/compose.config.js

```diff
@@ -0,0 +1,70 @@
+const { PREFIX, OUTPUT_PATH, OUTPUT_BASE_FILENAME } = process.env;
+const WHITELABEL = process.env.WHITELABEL !== 'false';
+
+const composeObjectTransformGroup = [
+    'color/composeColor',
+  ];
+
+function format(string){
+  const dashlessString = string.replace(/-(.)/g, (_, letter) => letter.toUpperCase());
+  const upperCasedString = dashlessString.charAt(0).toUpperCase() + dashlessString.slice(1);
+  const ampersandRemovedString = upperCasedString.replace(/&/g, 'And')
+  return ampersandRemovedString;
+}  
+  
+  
+module.exports = {
+    include: ['src/core/**/*.json5'],
+    source: [
+      ...(WHITELABEL === false ? ['src/telekom/core/**.json5'] : []),
+      'src/semantic/**/*.json5',
+    ],
+    platforms: {
+      composeObjectLightOnlyData: {
+        transforms: ['mode-light', ...composeObjectTransformGroup],
+        prefix: PREFIX,
+        buildPath: OUTPUT_PATH + 'android-compose/',
+        files: [
+          {
+            destination: OUTPUT_BASE_FILENAME + '.light.kt',
+            format: 'compose/object',
+            packageName: 'de.telekom.tokens',
+            className: 'ScaleTokensLight',
+            filter: (token) => {
+                if (token.path[0] !== 'core' && !token.path.includes('experimental') && token.type === 'color') {
+
+                  {
+                      const formatted = token.path.map(el => format(el))
+                      token.name = `Telekom${formatted.toString().replace(/,/g, '')}`
+                      return token
+                  }
+                }
+            }
+          },
+        ],
+      },
+      composeObjectDarkOnlyData: {
+        transforms: ['mode-dark', ...composeObjectTransformGroup],
+        prefix: PREFIX,
+        buildPath: OUTPUT_PATH + 'android-compose/',
+        files: [
+          {
+            destination: OUTPUT_BASE_FILENAME + '.dark.kt',
+            format: 'compose/object',
+            packageName: 'de.telekom.tokens',
+            className: 'ScaleTokensDark',
+            filter: (token) => {
+                if (token.path[0] !== 'core' && token.original.value?.dark != null && token.type === 'color')
+
+                {
+                    const formatted = token.path.map(el => format(el))
+                    token.name = `Telekom${formatted.toString().replace(/,/g, '')}`
+                    return token
+                }
```

Comment:
This looks like this filter function is being misused (it works nevertheless). The filter function should only do the checks and return a boolean.

For formatting the name you could register a [custom tranform of `type` "name"](https://amzn.github.io/style-dictionary/#/api?id=registertransform). I think it works the same was as value transform, you return the new name instead of setting it in the token object.

Reply:
thanks i was still a bit unsure about when/where to use transformers vs filters (even though the name should give it away)